clustermesh/clustermesh.go

```diff
@@ -2082,6 +2073,107 @@ func (k *K8sClusterMesh) ConnectWithHelm(ctx context.Context) error {
 	return nil
 }
 
+// Helm-based clustermesh enable is only supported on Cilium
+// v1.14+ due to a lack of support in earlier versions for
+// autoconfigured certificates (tls.{crt,key}) for cluster
+// members when running in certgen (cronJob) PKI mode
+func (k *K8sClusterMesh) inClassicMode(r *release.Release) (bool, error) {
+	version := r.Chart.AppVersion()
+	semv, err := utils.ParseCiliumVersion(version)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse Cilium version: %w", err)
+	}
+	k.Log("✅ Detected Helm release with Cilium version %s", semv)
+
+	const ciliumHelmMinRev = "1.14.0"
+	cv, err := semver.Parse(ciliumHelmMinRev)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse Cilium version: %w", err)
+	}
+	if cv.Compare(semv) < 0 {
+		k.Log("⚠️ Cilium Version is less than 1.14.0. Continuing in classic mode.")
+		return true, nil
+	}
```

Comment:
This will not work for pre-releases. I guess that it should be either restored the previous approach, or the version changed to something like 1.13.99 (don't know if there's a better way though). The comparison seems also to be inverted (it is checking if `1.14.0 < semver`). I'd suggest to use `LT/GT` rather than `Compare` for increased clarity.

Reply:
Good point! I'll take another pass. No keen on the whole string compare deal.